### PR TITLE
WTF-16443 Ability to create different callbacks with PATCH AND POST a…

### PR DESCRIPTION
**Pull request by mariane-bekker**
…ccording to resources passed in

### CODE REVIEW

**Comment by mariane-bekker** [comment]
Now, patch and put will automatically call the appropriate functions depending on the resource passed in.
EX: if user is patching age in about, then the function called will automatically be the function associated with age

File: test/lib/Router_test.js [hunk 1]
```diff
@@ -180,18 +180,159 @@ describe('API Router', function() {
...
-            .then(function(foo) {
-                this.getFooSpy.called.should.be.true;
-                var request = this.getFooSpy.lastCall.args[0];
-                request.should.be.instanceOf(Request);
-                request.getParam('fooId').should.equal('123');
+        describe('calls associated callbacks', function() {
+            it(' and connection objects with get method being a function', function(done) {
+                this.connection.get('/foo/123')
+                .then(function(foo) {
+                    console.log("SDSDS");
```
Comment (davidkaminsky) [inline test/lib/Router_test.js]: console.log

Comment (mariane-bekker) [inline test/lib/Router_test.js]: oops updated

**Comment by davidkaminsky** [comment]
😹

**Pull request merged**